bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -156,26 +161,110 @@ private void updateSchedulingBlockEnabledState(final boolean enabled) {
 
     private void updateDetailEditorEnabledState(final boolean enabled) {
         // Update enabled state for all detail widgets.
-        _w.detailEditor.allEditorsJava().stream().forEach(ed -> updateEnabledState(new Component[]{ed}, enabled));
+        _w.detailEditor.allEditorsJava().forEach(ed -> updateEnabledState(new Component[]{ed}, enabled));
+    }
+
+    private void updateCoordinateEditorEnabledState(final boolean enabled) {
+        updateEnabledState(new Component[]{_w.coordinateEditor}, enabled);
     }
 
     /**
-     * Simplifications for dealing with the Either representing selection.
+     * Simplifications for dealing with the current selection.
      */
     private Option<SPTarget> selectedTarget() {
-        return _curSelection.flatMap(t -> ImOption.fromOptional(t.toOptionalLeft()));
+        if (selection instanceof TableTargetSelection)
+            return new Some<>(((TableTargetSelection) selection).getTarget());
+        else
+            return None.instance();
     }
 
     private Option<IndexedGuideGroup> selectedGroup() {
-        return _curSelection.flatMap(g -> ImOption.fromOptional(g.toOptional()));
+        if (selection instanceof TableGroupSelection)
+            return new Some<>(((TableGroupSelection) selection).getGroup());
+        else
+            return None.instance();
+    }
+
+    private Option<SPCoordinates> selectedCoordinates() {
+        if (selection instanceof TableCoordinateSelection)
+            return new Some<>(((TableCoordinateSelection) selection).getCoordinates());
+        else
+            return None.instance();
     }
 
     /**
-     * Auxiliary method to determine if the current selection is the base position.
+     * Extract the base position out of the asterism.
+     * It could be a target or a sky position.
+     * Due to type inference issues with map, we have to do this in a horrible way.
      */
-    private boolean selectionIsBasePosition() {
+    private ImEither<SPTarget, SPCoordinates> basePosition() {
         final TargetEnvironment env = getDataObject().getTargetEnvironment();
-        return selectedTarget().exists((target) -> target == env.getArbitraryTargetFromAsterism());
+        final Asterism a            = env.getAsterism();
+
+        switch (a.asterismType()) {
+            case Single:
+                return new Left<>(((Asterism.Single) a).t());
+            case GhostSingleTarget:
+                final GhostAsterism.SingleTarget gsa = (GhostAsterism.SingleTarget) a;
+                if (gsa.base().isDefined()) return new Right<>(gsa.base().get());
```

Comment:
Maybe we should just commit to fixing any issues that come up for November and put it in `develop`?  Otherwise we'd run into issues with having multiple branches that we have to keep in sync.  I guess that does mean that the Ghost testing run in September comes with a bunch of other unrelated and untested changes though.  @arturog8m what do you think?

Reply:
I'd prefer this, if everyone agrees. Given the scope of these changes, trying to merge at a later date is likely to cause serious merge conflicts.

Reply:
> I'd prefer this, if everyone agrees. Given the scope of these changes, trying to merge at a later date is likely to cause serious merge conflicts.

It needs to go into `develop` regardless so I don't see a reason not to do that now.

Reply:
The `ImEither` approach is much cleaner. Thanks for that recommendation. This particularly hideous block of code with all the `isDefined()`...`get()` was not sitting well with me.